frontends/mrxl/fud/mrxl.py

```diff
@@ -56,5 +57,72 @@ def run_mrxl(mrxl_prog: SourceType.Path) -> SourceType.Stream:
         return run_mrxl(input)
 
 
+class MrXLDataStage(Stage):
+    """
+    Stage that invokes the MrXL frontend.
+    """
+
+    name = "mrxl-data"
+
+    def __init__(self):
+        """
+        Initialize this stage. Initializing a stage *does not* construct its
+        computation graph.
+        """
+        super().__init__(
+            src_state="mrxl-data",
+            target_state="verilog-data",
+            input_type=SourceType.Path,
+            output_type=SourceType.Stream,
+            description="Compiles MrXL-native input to Calyx-native input.",
+        )
+
+    @staticmethod
+    def pre_install():
+        pass
+
+    @staticmethod
+    def defaults():
+        """
+        Specify defaults that should be added to fud's configuration file when
+        this stage is registered.
+        """
+        return {"exec": "mrxl-data"}
+
+    def _define_steps(self, input, builder, config):
+        """
+        Define the steps that will execute in this stage. Each step represents
+        a delayed computation that will occur when the stage is executed.
+        """
+
+        # Commands at the top-level are evaluated when the computation is being
+        # staged
+        cmd = config["stages", "mrxl", "exec"]
+        data_path = config["stages", "mrxl", "data"]
```

Comment:
The reason you get the error message is because the code here attempts to get the value in the configuration. If the key is not found, the configuration will error out. If you want to avoid the error, use the `.get` method instead:
```
data_path = config.get(["stages", "mrxl", "data"])
```